src/librustc_mir/build/mod.rs

```diff
@@ -341,6 +341,11 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             }
         }
     }
+
+    pub fn check_overflow(&self) -> bool {
```

Comment:
Hmm. So I think that @arielb1 and @Aatch are right about how things should work, but I also think @eddyb is right that we want e.g. `pow` to overflow ideally. The way I think this ideally would be happening is that we have "debug builds" of core/std that have checks enabled. But that's obviously a big thing. So I'm not sure what's best here. If we wanted to keep the "check for inlined code" behavior for the time being, we have to make some transformation as part of lowering for trans I guess, and presumably we'd export a "pre-lowering" version for inlining? It's all kind of a big mess.

I guess we could also build the checks into MIR more and do the choice in trans, but that will generate suboptimal MIR (since we must accommodate panics etc).

Not sure what's best strategy here, I have to admit.